tests/performance/density-kwok.go

```diff
@@ -0,0 +1,343 @@
+/*
+Copyright 2024 The KubeVirt Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package performance
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/libvmifact"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+const (
+	vmBatchStartupLimit = 5 * time.Minute
+	defaultNodeCount    = 100
+	defaultVMCount      = 1000
+)
+
+var (
+	nodeCount = getNodeCount()
+	vmCount   = getVMCount()
+)
+
+var _ = Describe("Control Plane Performance Density Testing using kwok", Label("KWOK", "sig-performance", "Serial"), Serial, Ordered, func() {
```

Comment:
`Seria`l is already present:
```suggestion
var _ = Describe("Control Plane Performance Density Testing using kwok", Label("KWOK", "sig-performance"), Serial, Ordered, func() {
```

Usually, we want to avoid the usage of `Ordered` since every test should be independent each other.
Just a suggestion: why don't we structure the test file as the following?
```
	Describe("kwok density tests", func() {
		BeforeEach(func() {
			// Create the fake nodes
		})

		AfterEach(func() {
			// Delete the nodes
		})

		Context(fmt.Sprintf("create a batch of %d fake VMIs and VMs", vmCount), func() {
			It("should successfully create all fake VMIs and VMs", func() {
				// VMI block (create/delete/collect metrics)

				//wait for 5 mins to bring the metrics to steady state
				time.Sleep(5 * time.Minute)

				// VM block (create/delete/collect metrics)
			})
		})
	})
```
In this way we can get rid of the `Ordered`

Reply:
We wanted to have this workflow in our tests, creating and deleting the nodes only once
https://github.com/kubevirt/kubevirt/blob/ce4a7bf884d6a92ebc308ffd9b52f55400c39d1a/tests/performance/README.md#workflow-of-the-kwok-e2e-test

Reply:
> var _ = Describe("Control Plane Performance Density Testing using kwok", Label("KWOK", "sig-performance"), Serial, Ordered, func() 

This will not take Serial as a label right? since its a test modifier

Reply:
The `Serial` is enough to make them run in serial

Reply:
> We wanted to have this workflow in these tests, creating and deleting the nodes only once https://github.com/kubevirt/kubevirt/blob/ce4a7bf884d6a92ebc308ffd9b52f55400c39d1a/tests/performance/README.md#workflow-of-the-kwok-e2e-test

If we really cannot get rid of the `Ordered` I'd suggest you to use `BeforeAll` and `AfterAll` instead of respectively the first and the second `It`.
Honestly I still think that it is not required, but I will not block this

Reply:
@fossedihelm  
We can setup fake nodes in SynchronizedBeforeSuite by checking if kwoktests env is enabled or not. Do you have any concerns on this?

Reply:
Is there a decorator for `sig-performance`? Can you create another decorator for `KWOK`?

Reply:
I have created a decorator, PTAL